pyvista/core/cell.py

```diff
@@ -510,3 +518,113 @@ def cells(self):
     def n_cells(self):
         """Return the number of cells."""
         return self.GetNumberOfCells()
+
+    @property
+    def connectivity_array(self):
+        """Return the array with the point ids that define the cells' connectivity."""
+        return _get_connectivity_array(self)
+
+    @property
+    def offset_array(self):
+        """Return the array used to store cell offsets."""
+        return _get_offset_array(self)
+
+    @staticmethod
+    def from_arrays(offsets, connectivity, deep=False) -> 'CellArray':
+        """Construct a vtkCellArray from offsets and connectivity arrays.
+
+        Parameters
+        ----------
+        offsets : numpy.ndarray or list[int]
+            Offsets array of length `n_cells + 1`
+
+        connectivity : numpy.ndarray or list[int]
+            Connectivity array
+
+        deep : bool, default: False
+            Whether to deep copy the array data into the vtk arrays.
+            Default is ``False``.
+        """
+        cellarr = CellArray()
+        offsets = numpy_to_idarr(offsets, deep=deep)
+        connectivity = numpy_to_idarr(connectivity, deep=deep)
+        cellarr.SetData(offsets, connectivity)
+        return cellarr
+
+    @property
+    def regular_cells(self) -> np.ndarray:
+        """Return an array of shape (n_cells, cell_size) of point indices when all faces have the same size.
+
+        Returns
+        -------
+        numpy.ndarray
+            Array of face indices of shape (n_cells, cell_size).
+
+        Notes
+        -----
+        This property does not validate that the cells are all
+        actually the same size. If they're not, this property may either
+        raise a `ValueError` or silently return an incorrect array.
+        """
+        return _get_regular_cells(self)
+
+    @classmethod
+    def from_regular_cells(cls, cells, deep=True):
+        """Constract a ``CellArray`` from a (n_cells, cell_size) array of cell indices.
+
+        Parameters
+        ----------
+        cells : numpy.ndarray or list[list[int]]
+            Cell array of shape (n_cells, cell_size) where all cells have the same size `cell_size`
+
+        deep : bool, default: False
```

Comment:
Does this work without copied data?  That is if I pass in `cells` and then modify `cells` will it work as intended?  Is there a problem if `cells` changes size, which can happen for a `list[list[int]]`?

Reply:
No, see below.